### ISSUE TITLE
cloud mode crushes logics

### ISSUE THREAD

**Issue by gribouk** [AlexxIT/SonoffLAN#292]
Hello!
Is it possible to make the devices work in local mode, which support local mode, and in cloud mode, the devices supporting cloud mode only?

The issue: as of this midnight, I belive, the Chinise cloud experiences some issues again, which can be discovered through the fact, that it does not update itself on device state update: for example, you press a button on a 4CH, it turns on a channel, but state in eWelink does not change, maybe once out of 10 times it reacts, but the other times fails. As the result, when my "advanced logics" scripts  in HA do certain computations and change the state of 4CH, the cloud does not follow. Instead, after some time, it decides to update my entities in HA (or vice versa, HA wants to synchronise itself), and changes their state to the ones, it refused to change already (the wrong ones to that moment)! And all my 'advanced logics' crushes... I spent whole day investigatig the cause of the sudden malfunction which was not there  yesterday evening before discovering this. With local mode in the setup I loose zigbee temperature sensors which data I need for "advanced logics"! Hence the question: is it possible to start zigbee in cloud and the rest of the field in local modes?

**Comment by AlexxIT** [comment]
What's your current config?

**Comment by gribouk** [comment]
This is eWelink's part only:

```
########################## EWeLink Section BEGIN ####################################        
sonoff:
  username: gribouk@gmail.com
  password: 192837465qazwsx
##  mode: local
##  reload: always  # update device list every time HA starts  
  
  sensors: [temperature, humidity, power, current, voltage]
  force_update: [temperature, humidity, power, current, voltage]
  scan_interval: '00:00:10'  # (optional) default 5 minutes
  
  
  
  
  rfbridge:
    dining1:  # sensor name in eWeLink application
      device_class: motion
      timeout: 120 
    kitchen1:  # sensor name in eWeLink application
      device_class: motion
      timeout: 120 
    stairs_to_spa:  # sensor name in eWeLink application
      device_class: motion
      timeout: 20
    stairs_up_second_floor:  # sensor name in eWeLink application
      device_class: motion
      timeout: 20 
    stairs_down_first_floor:  # sensor name in eWeLink application
      device_class: motion
      timeout: 20 
    library:  # sensor name in eWeLink application
      device_class: motion
      timeout: 120
    first_floor_wc:  # sensor name in eWeLink application
      device_class: motion
      timeout: 120 
    gym1:  # sensor name in eWeLink application
      device_class: motion
      timeout: 120 
    laundry:  # sensor name in eWeLink application
      device_class: motion
      timeout: 120 
    electricity:  # sensor name in eWeLink application
      device_class: motion
      timeout: 120
    boiler_room:  # sensor name in eWeLink application
      device_class: motion
      timeout: 120
    spa_massage:  # sensor name in eWeLink application
      device_class: motion
      timeout: 120
    spa_wc:  # sensor name in eWeLink application
      device_class: motion
      timeout: 120
    spa_shawer:  # sensor name in eWeLink application
      device_class: motion
      timeout: 120
    basement_corridor:  # sensor name in eWeLink application
      device_class: motion
      timeout: 20 
    basement_corridor1:  # sensor name in eWeLink application
      device_class: motion
      timeout: 20 
    wardrobe:  # sensor name in eWeLink application
      device_class: motion
      timeout: 120 
    corridor_second_floor1:  # sensor name in eWeLink application
      device_class: motion
      timeout: 20
    master_wc:  # sensor name in eWeLink application
      device_class: motion
      timeout: 120
    guest_wc1:  # sensor name in eWeLink application
      device_class: motion
      timeout: 120 
    guest_wc2:  # sensor name in eWeLink application
      device_class: motion
      timeout: 120
    master_wardrobe:  # sensor name in eWeLink application
      device_class: motion
      timeout: 120 
    master_corridor:  # sensor name in eWeLink application
      device_class: motion
      timeout: 20
    storage:  # sensor name in eWeLink application
      device_class: motion
      timeout: 120 

########################### EWeLink Section END ################################ 

```

**Comment by gribouk** [comment]
As of this moment, I had to delete all devices that did not synchronise with eWelenk on button action and readd those again to make them work as yesterday. Urgent metter solved, but the issue remains: is it possible to do something about it in more generalized approach? Options are:

1. Set synchro rules priorities: HA above eWelink or vice versa.
2. Set device dependant mode: zigbee cloud, the rest local.

Any of the above solves the problem. I cannot depend on Chinise cloud behavior because it is unpredictible. Almost every week they have major crush of something... This time it only concerned my 4CH - all of them could not send state update to the cloud, while cloud still send commands to them. As the result, HA in a very unpredictible way picks the state from the cloud which was changed long ago localy...

**Comment by AlexxIT** [comment]
Have you seen this problem in the logs? How did you understand that the wrong state from the cloud overwrite the correct state from the LAN?

**Comment by gribouk** [comment]
Could it be this one:
2020-11-26 20:36:29 ERROR (MainThread) [custom_components.sonoff.sonoff_cloud] Cloud WS exception: {'action': 'update', 'deviceid': '1000d6169f', 'apikey': 'd4f1de5b-0156-49b9-9a1d-6a3e60035f0f', 'userAgent': 'device', 'sequence': '1606415787986', 'ts': 0, 'params': {'sledOnline': 'on'}, 'from': 'device', 'seq': '4074'}

But I did not look into the log before. I was catching it analytically: I debugged the scripts before I understood it was not about the scripts but abnormal device behaviour - it changed back it's state while no trigger fired. Then I realized that these devices have buttons, and if I disconnect the HA from the network and press the button, it would be only device and cloud communication - and I discovered that I can change the state of 4CH from the cloud, but cannot change the state in the cloud if button pressed and device has changed it's state actually. Then I tested local mode and confirmed loosing zigbees, then I wrote an issue...
